### PR TITLE
added word embedding visualisation in docs/word2vec.ipynb

### PR DESCRIPTION
**Pull request by aneesh-joshi**
made the changes as per issue https://github.com/RaRe-Technologies/gensim/issues/1419
Some changes are there due to opening notebook in my version of python and ipython
If required, will change that.
Please review.

Major changes are at the end of the notebook in /docs/word2vec.ipynb towards the end

See the issue to get a quick idea of how the embeddings look

### CODE REVIEW

File: docs/notebooks/word2vec.ipynb [hunk 1]
```diff
@@ -1082,8 +1079,5322 @@
...
+    "import numpy as np                                  # array handling\n",
+    "\n",
+    "from plotly.offline import init_notebook_mode, iplot, plot\n",
+    "import plotly.graph_objs as go\n",
+    "\n",
+    "def reduce_dimensions(model, plot_in_notebook = True):\n",
+    "\n",
+    "    num_dimensions = 2        # final num dimensions (2D, 3D, etc)\n",
+    "\n",
+    "    vectors= []         # positions in vector space\n",
```
Comment (menshikh-iv) [inline docs/notebooks/word2vec.ipynb]: Add space before `=`

File: docs/notebooks/word2vec.ipynb [hunk 2]
```diff
@@ -1082,8 +1079,5322 @@
...
+    "    labels = []         # keep track of words to label our data again later\n",
+    "    for word in model.wv.vocab:\n",
+    "        vectors.append(model[word])\n",
+    "        labels.append(word)\n",
+    "\n",
+    "\n",
+    "    # convert both lists into numpy vectors for reduction\n",
+    "    vectors = np.asarray(vectors)\n",
+    "    labels =  np.asarray(labels)\n",
```
Comment (menshikh-iv) [inline docs/notebooks/word2vec.ipynb]: One space after `=`

File: docs/notebooks/word2vec.ipynb [hunk 3]
```diff
@@ -1082,8 +1079,5322 @@
...
+    "    \n",
+    "    # reduce using t-SNE\n",
+    "    vectors = np.asarray(vectors)\n",
+    "    logging.info('starting tSNE dimensionality reduction. This may take some time.')\n",
+    "    tsne = TSNE(n_components=num_dimensions, random_state=0)\n",
+    "    vectors = tsne.fit_transform(vectors)\n",
     "\n",
+    "    x_vals = [ v[0] for v in vectors ]\n",
```
Comment (menshikh-iv) [inline docs/notebooks/word2vec.ipynb]: No needed space after `[` and before `]` (same in next line)

File: docs/notebooks/word2vec.ipynb [hunk 4]
```diff
@@ -1082,8 +1079,5322 @@
...
+    "    y_vals = [ v[1] for v in vectors ]        \n",
+    "        \n",
+    "    # Create a trace\n",
+    "    trace = go.Scatter(\n",
+    "        x = x_vals,\n",
```
Comment (menshikh-iv) [inline docs/notebooks/word2vec.ipynb]: No needed space before/after `=`, it's function kwargs (same in next 3 lines)

File: docs/notebooks/word2vec.ipynb [hunk 5]
```diff
@@ -1082,8 +1079,5322 @@
...
+   "cell_type": "code",
+   "execution_count": 49,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2017-06-22 11:06:03,601 : INFO : starting tSNE dimensionality reduction. This may take some time.\n",
+      "2017-06-22 11:06:11,212 : INFO : All done. Plotting.\n"
```
Comment (menshikh-iv) [inline docs/notebooks/word2vec.ipynb]: Please add description for "clusters" (why do words fall into one / different clusters?).

File: docs/notebooks/FastText_Tutorial.ipynb [hunk 6]
```diff
@@ -300,8 +300,8 @@
       ]
      },
      "execution_count": 7,
-     "output_type": "execute_result",
-     "metadata": {}
+     "metadata": {},
```
Comment (aneesh-joshi) [inline docs/notebooks/FastText_Tutorial.ipynb]: Sorry.
I had opened it up in jupyter which made the changes to it automatically.
removed it.

File: docs/notebooks/gensim Quick Start.ipynb [hunk 7]
```diff
@@ -51,17 +47,16 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-     "This is a particularly small example of a corpus for illustration purposes. Another example could be a list of all the plays written by Shakespeare, list of all wikipedia articles, or all tweets by a particular person of interest.\n",
+    "This is a particularly small example of a corpus for illustration purposes. Another example could be a list of all the plays written by Shakespeare, list of all wikipedia articles, or all tweets by a particular person of interest.\n",
```
Comment (aneesh-joshi) [inline docs/notebooks/gensim Quick Start.ipynb]: Hi
I couldn't figure out how to remove it form my PR
so I just undid the change I had made

Comment (aneesh-joshi) [inline docs/notebooks/gensim Quick Start.ipynb]: removed the change I had made.

**Comment by aneesh-joshi** [comment]
If this is too cluttered, I could close this PR and make a fresh one...

**Comment by menshikh-iv** [comment]
@aneesh-joshi Sorry for late answer. You no need to create new PR, please remove changes with all files (except `word2vec.ipynb`) & resolve conflicts in `word2vec.ipynb`, after this I think I will merge this PR

**Comment by menshikh-iv** [comment]
Ping @aneesh-joshi, what's a status of PR?

**Comment by menshikh-iv** [comment]
I close PR because it's abandoned.

**Comment by piskvorky** [comment]
@menshikh-iv please add an issue label to remember to work on this ourselves. If I understand correctly, this feature is nearly finished, just some final refactoring missing.

**Comment by aneesh-joshi** [comment]
Hi
Sorry about that. I'm a bit busy.
Will do it soon.

**Comment by menshikh-iv** [comment]
@aneesh-joshi please re-open PR when you'll be ready and ping me.

**Comment by menshikh-iv** [comment]
Finished in #1800
